File: week1/greed/less/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    // Initializing variables
    int changeOwed = -1;
    int coin[] = {25, 10, 5, 1};
    int remaning, amount = 0;
    // Prompt for Change owed
    while (changeOwed < 0)
    {
        changeOwed = round(100 * get_float("Change owed: "));
    }
    // Generating Change
    for (int i = 0; i < 4; i++)
    {
        amount += changeOwed / coin[i];
        remaning = changeOwed / coin[i];
        changeOwed -= coin[i] * remaning;
    }
    // Print amount
    printf("%d\n", amount);
}

File: week6/greed/less/cash.py
from cs50 import get_float

# Setup varibles
coins = [25, 10, 5, 1]
changeOwed = -1
remaning = 0
amount = 0

# Get change
while changeOwed < 0:
    changeOwed = round(100 * get_float("Change owed: "))

# Setup amount
for coin in coins:
    amount += changeOwed // coin
    remaning = changeOwed // coin
    changeOwed -= coin * remaning

# Print result
print(amount)
